src/main/java/seedu/address/logic/commands/AddPersonCommand.java

```diff
@@ -132,10 +131,26 @@ public boolean equals(Object other) {
         }
 
         AddPersonCommand otherCommand = (AddPersonCommand) other;
-        return name.equals(otherCommand.name)
-                && phone.equals(otherCommand.phone)
-                && email.equals(otherCommand.email)
-                && Objects.equals(internshipId, otherCommand.internshipId)
-                && tags.equals(otherCommand.tags);
+        if (phone == null && email == null) {
+            return name.equals(otherCommand.name)
+                    && Objects.equals(internshipId, otherCommand.internshipId)
+                    && tags.equals(otherCommand.tags);
+        } else if (phone == null) {
+            return name.equals(otherCommand.name)
+                    && email.equals(otherCommand.email)
+                    && Objects.equals(internshipId, otherCommand.internshipId)
+                    && tags.equals(otherCommand.tags);
+        } else if (email == null) {
+            return name.equals(otherCommand.name)
+                    && phone.equals(otherCommand.phone)
+                    && Objects.equals(internshipId, otherCommand.internshipId)
+                    && tags.equals(otherCommand.tags);
+        } else {
+            return name.equals(otherCommand.name)
+                    && email.equals(otherCommand.email)
+                    && phone.equals(otherCommand.phone)
+                    && Objects.equals(internshipId, otherCommand.internshipId)
+                    && tags.equals(otherCommand.tags);
+        }
```

Comment:
We can simplify this using `Objects.equals` since it accounts for null.
```
return name.equals(otherCommand.name)
                && Objects.equals(phone, otherCommand.phone)
                && Objects.equals(email, otherCommand.email)
                && Objects.equals(internshipId, otherCommand.internshipId)
                && tags.equals(otherCommand.tags);
```